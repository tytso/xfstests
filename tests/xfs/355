#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2017 Oracle, Inc.  All Rights Reserved.
#
# FS QA Test No. 355
#
# Populate a XFS filesystem and fuzz every AGFL field.
# Use xfs_scrub to fix the corruption.
#
. ./common/test_names
_set_seq_and_groups dangerous_fuzzers dangerous_scrub dangerous_online_repair

trap "_cleanup; exit \$status" 0 1 2 3 7 15

_cleanup()
{
	cd /
	rm -rf $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_fs xfs
_require_scratch_xfs_fuzz_fields

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Fuzz AGFL"
_scratch_xfs_fuzz_metadata '' 'online' 'agfl 0' >> $seqres.full
echo "Done fuzzing AGFL"

echo "Fuzz AGFL flfirst"
flfirst=$(_scratch_xfs_db -c 'agf 0' -c 'p flfirst' | sed -e 's/flfirst = //g')
SCRATCH_XFS_LIST_METADATA_FIELDS="bno[${flfirst}]" _scratch_xfs_fuzz_metadata '' 'online' 'agfl 0' >> $seqres.full
echo "Done fuzzing AGFL flfirst"

# success, all done
status=0
exit
