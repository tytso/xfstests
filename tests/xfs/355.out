QA output created by 355
Format and populate
Fuzz AGFL
scrub didn't fail with lsn = zeroes.
scrub didn't fail with lsn = ones.
scrub didn't fail with lsn = lastbit.
scrub didn't fail with lsn = sub.
scrub didn't fail with bno[0] = zeroes.
scrub didn't fail with bno[0] = ones.
scrub didn't fail with bno[0] = middlebit.
scrub didn't fail with bno[0] = lastbit.
scrub didn't fail with bno[0] = add.
Done fuzzing AGFL
Fuzz AGFL flfirst
scrub didn't fail with bno[1] = zeroes.
scrub didn't fail with bno[1] = ones.
scrub didn't fail with bno[1] = middlebit.
scrub didn't fail with bno[1] = lastbit.
scrub didn't fail with bno[1] = add.
Done fuzzing AGFL flfirst
